### PR TITLE
Bluetooth: Mesh: Remaining time should not include Delay time

### PR DESCRIPTION
**Pull request by michal-narajowski**
Fix according to MMDL Errata 16106.

Signed-off-by: Michał Narajowski <michal.narajowski@codecoup.pl>

### CODE REVIEW

**Comment by michal-narajowski** [comment]
Fixes MMDL/SR/LLC/BV-04-C after CASE0071677.

**Review by trond-snekvik — APPROVED** [review]

**Comment by trond-snekvik** [comment]
Corresponding test PR posted for Mesh tests.

**Comment by PavelVPV** [comment]
I will align system tests along with the errata.

I think there is one corner case here, which looks a bit awkward, but needs to be covered as well. While light is off, if you send Light OnOff Set message with Transition Time set to 0 and Delay set to non zero value, Light OnOff Status message (in current implementation) will have the following body: 'present': False, `target`: `True`, 'remaining_time': `0`. But if you manage to send Generic OnOff Get message during the delay, Generic OnOff server will respond with Status message where only `present` value is present in the message because the remaining time is zero: https://github.com/nrfconnect/sdk-nrf/blob/5c523e77abd79fae1210188a0a1bbd5f940a4d2c/subsys/bluetooth/mesh/gen_onoff_srv.c#L13-L24
which is correct since state change has not been executed yet.

Since the the state change in Light LC server has not been executed due to delay, I assume that Light OnOff Status message should replay with only `present` value.

**Review by omkar3141 — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]
Discussed offline with @omkar3141. 

As per my understanding, the following part of E16106 introduced inconsistency into the spec (section 1.4.1):

> There are also messages reporting both the present value of a state and the target value of the state, along with the remaining time, which is the time it will take from the moment the message is sent by the model to the end of the transition to the target state (if the transition has already started), or **the transition time (if the transition is yet to be started).** 

Because this contradicts with what _every_ server defines for Status message behaviour. E.g. for Generic OnOff Server (section 3.3.1.2.3):

> When sending a Generic OnOff Status message, the Generic OnOff Server shall set the OnOff field to the present Generic OnOff state. **If the Generic OnOff Server is in the process of changing the Generic OnOff state**, it shall set the Target OnOff field to the target Generic OnOff state and shall set the Remaining Time field to the time it will take to complete the transition; otherwise, the Target Level and Remaining Time fields shall be omitted. 

It was pointed out by @omkar3141 that a Status message sent as a respond to a Set message always has `present`, `target` and `remaining time` fields present no matter of whether the delay field was present or not. This, however, is _not_ specified anywhere in the spec and more over contradicts with what spec states for _every_ Set message behaviour as there is _no_ difference between Status message sent as a response and as unsolicited message. See as an example section 3.3.1.2.2:

> If the received message is a Generic OnOff Set message, the Generic OnOff Server shall respond with a Generic OnOff Status message (see Section 3.3.1.2.3).

Current implementation including change in the PR looks fine. It works for the case where Light OnOff Set message is sent with non-zero transition time and non-zero delay. Then Light OnOff Status message will contain all optional fields (even though this doesn't comply with section 6.5.1.6.3). _But_ if Generic OnOff state is bound with Light LC Light OnOff state and if Generic OnOff Set message is sent with zero transition time and non-zero delay, Generic OnOff Status message will **only** have `present` field in the body, which does not comply with the text above: https://github.com/nrfconnect/sdk-nrf/blob/5c523e77abd79fae1210188a0a1bbd5f940a4d2c/subsys/bluetooth/mesh/gen_onoff_srv.c#L13-L24

Basically, every other model (except Light LC) doesn't consider that the behaviour for the Status message sent as a reply is different from an unsolicited Status message). This means that other models needs to be aligned with this logic.

Even though to fix current implementation we need to change behaviour of a response (basically, fix `rsp_status` function in every generic model and same functions in other models), we have decided to make our system tests _ugly_ just to follow current implementation and fix the models implementation later after the release as this corner case is _not_ covered by the test spec.

**Comment by trond-snekvik** [comment]
CI is passing in a rerun of the mesh test suite.
